### PR TITLE
ci: add LLVM support

### PR DESCRIPTION
**Pull request by mykola-lysenko**
Signed-off-by: Mykola Lysenko <mykolal@fb.com>

### CODE REVIEW

File: .github/actions/vmtest/build_selftests.sh [hunk 1]
```diff
@@ -6,9 +6,19 @@ THISDIR="$(cd $(dirname $0) && pwd)"
 
 source "${THISDIR}"/helpers.sh
 
-travis_fold start prepare_selftests "Building selftests"
+TOOLCHAIN="$1"
+TOOLCHAIN_NAME="$(echo $TOOLCHAIN | cut -d '-' -f 1)"
+TOOLCHAIN_VERSION="$(echo $TOOLCHAIN | cut -d '-' -f 2)"
 
+if [ "$TOOLCHAIN_NAME" == "llvm" ]; then
+export LLVM="-$TOOLCHAIN_VERSION"
```
Comment (anakryiko) [inline .github/actions/vmtest/build_selftests.sh:14]: isn't it always LLVM=1 ?

Comment (anakryiko) [reply .github/actions/vmtest/build_selftests.sh:14]: btw, @yonghong-song, FYI, we'll soon start detecting all these LLVM-specific issues early on. Please take a look at this as well.

**Review by anakryiko — COMMENTED** [review]
looks good (not sure about LLVM=15, though), but let's wait for my fix to land first

File: .github/actions/vmtest/build_selftests.sh (hunk 1 above)
Comment (mykola-lysenko) [reply .github/actions/vmtest/build_selftests.sh:14]: LLVM toolchain we install in CI has version suffix (currently it is 15).

Linux Kernel build system handles this gracefully: https://github.com/torvalds/linux/blob/b4a5ea09b29371c2e6a10783faa3593428404343/tools/scripts/Makefile.include#L54-L73

I do not like however that it still defaults to clang without version suffix: https://github.com/torvalds/linux/blob/b4a5ea09b29371c2e6a10783faa3593428404343/tools/scripts/Makefile.include#L87-L92. This functionality is used when we build bpf c code in selftests (https://github.com/kernel-patches/vmtest/pull/82/files#diff-c34c251202780f09ab91ab0021a7190227e9a48d8063b004a644fe65a365125fR37-R39).

I am thinking about sending a patch to Linux build system to properly initialize CLANG variable when LLVM variable is set and has a version suffix provided

Comment (mykola-lysenko) [reply .github/actions/vmtest/build_selftests.sh:14]: Some more details in https://www.kernel.org/doc/html/latest/kbuild/llvm.html

Comment (anakryiko) [reply .github/actions/vmtest/build_selftests.sh:14]: nice, didn't know about it.

**Review by anakryiko — APPROVED** [review]

File: .github/actions/vmtest/build_selftests.sh (hunk 1 above)
Comment (yonghong-song) [reply .github/actions/vmtest/build_selftests.sh:14]: Looks like this llvm suffix thing is just extended two months ago:
  https://lore.kernel.org/all/20200317215515.226917-1-ndesaulniers@google.com/
the llvm prefix thing is extended earlier.

I don't have a strong opinion on this as I always use the default path 'clang' only. Whatever it is best for CI is also okay for me.

**Pull request merged**